plugins/nexus-repository-conda/src/main/java/org/sonatype/nexus/repository/conda/internal/hosted/CondaPath.java

```diff
@@ -0,0 +1,277 @@
+     */
+    @Nonnull
+    public CondaPath subordinateOf() {
+        return hashType.map(
+                currHash -> {
+                    int hashSuffixLen = currHash.getExt().length() + 1; // the dot
+                    Optional<Coordinates> mainCoordinates = coordinates.map(currCoordinates -> new Coordinates(
+                            currCoordinates.getPackageName(),
+                            currCoordinates.getVersion(),
+                            currCoordinates.getBuildString(),
+                            currCoordinates.getExtension().substring(0, currCoordinates.getExtension().length() - hashSuffixLen)
+                    ));
+                    String subPath = path.substring(0, path.length() - hashSuffixLen);
+                    return new CondaPath(subPath, mainCoordinates);
+                }
+        ).orElse(this);
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof CondaPath)) {
+            return false;
+        }
+        CondaPath that = (CondaPath) o;
+        return path.equals(that.path);
+    }
+
+    @Override
+    public int hashCode() {
+        return path.hashCode();
+    }
+
+    @Override
+    public String toString() {
+        return getClass().getSimpleName() + "{" +
+                "path='" + path + '\'' +
+                ", fileName='" + fileName + '\'' +
+                ", hashType=" + hashType +
+                '}';
+    }
+
+
+    /**
+     * Returns path of passed in hash type that is subordinate of this path. This path cannot be hash.
+     */
+    @Nonnull
+    public CondaPath hash(final HashType hashType) {
+        checkNotNull(hashType);
+        checkArgument(this.hashType.isPresent(), "This path is already a hash: %s", this);
+        Optional<Coordinates> hashCoordinates = coordinates.map(currCoordinates -> new Coordinates(
+                currCoordinates.getPackageName(),
+                currCoordinates.getVersion(),
+                currCoordinates.getBuildString(),
+                currCoordinates.getExtension() + "." + hashType.getExt()
+        ));
+        return new CondaPath(path + "." + hashType.getExt(), hashCoordinates);
+    }
+
+    private static Optional<Coordinates> parseCoordinates(String path) {
+
+        if (path.endsWith(REPODATA_JSON)) {
+            return Optional.empty();
+        }
+        int vEndPos = path.lastIndexOf('/');
+        if (vEndPos == -1) {
+            return Optional.empty();
+        }
+
+        final String fileName = path.substring(vEndPos + 1);
+        String[] parts = fileName.split("-");
+
+        String packageName = parts[0];
+        String version = "";
+        String build = "";
+        String extension = "";
+        if (parts.length > 1) {
+            version = parts[1];
+            build = parts[2].replace(".tar.bz2", "");
```

Comment:
Lot of magic numbers here and it's kind difficult to understand. Let's put into some named variables.

Reply:
this is borrowed code from https://github.com/kbc-opensource/nexus-conda-plugin/blob/75547852e17e860ba835abe118434fc73c519e6c/src/main/java/be/kbc/eap/nexus/internal/CondaPathParserImpl.java

It needs to be re-written from scratch - it is really hard to tell what is happening here.